### ISSUE TITLE
White space are missing when new line is started in html <p> tag

### ISSUE THREAD

**Issue by Saad1379** [Aymkdn/html-to-pdfmake#141]
@Aymkdn  Is there any way to resolve this issue? White space is missing when new line started in html
![image](https://user-images.githubusercontent.com/16044705/172923249-31e200b2-d5a5-4269-b78a-aed65c93d878.png)

**Comment by Aymkdn** [comment]
You need CSS to render it in HTML, so it's the same here:
```html
<p style="white-space:break-spaces">
    This
is
a
sentence</p>
```
![image](https://user-images.githubusercontent.com/946315/173027606-fecb0d0f-518e-4bf8-aa92-b528a9f13f3e.png)

**Comment by Saad1379** [comment]
@Aymkdn Actually this is not my problem. My problem is that I want to add white space between last word of a line and first word of next line. Currently they appear joined together.
![image](https://user-images.githubusercontent.com/16044705/173187313-6ffab823-33f0-4dd0-a574-2680d1fd5b8d.png)

![image](https://user-images.githubusercontent.com/16044705/173187282-f10bbd1a-892a-4d17-96af-ef3210d46139.png)

**Comment by Aymkdn** [comment]
What is the HTML code that you use?

**Comment by Saad1379** [comment]
This is the html code
`            <div class="section1 pages" id="section1">
                <h2 style="font-size: 18px; font-weight: 700;  color: #0b6ab0; text-align: center;">
                    Heading
                </h2>
                <p style="font-size: 14px; font-weight:300;">
                    Lorem ipsum dolor, sit amet consectetur adipisicing elit. Voluptate aliquid vel vero numquam
                    molestiae quidem, dolorum provident non minima illum tenetur aspernatur. Ratione perspiciatis
                    voluptates id placeat totam esse veniam!
                </p>
                <p style="font-size: 14px; font-weight:300">
                    Lorem ipsum dolor sit amet consectetur adipisicing elit. Porro consectetur unde totam consequuntur
                    vitae. Quis deserunt accusamus maiores veniam earum. Labore vitae iusto dolor corrupti veritatis
                    aperiam id voluptatem numquam!
                </p>
</div>`

**Comment by Aymkdn** [comment]
I'm a bit confused. What is the HTML code you use, how does it render in the browser, how it looks like in the pdf?

**Comment by Saad1379** [comment]
You can see the HTML code and the generated pdf in the following screenshot:
![image](https://user-images.githubusercontent.com/16044705/173199962-f228218d-9a74-410e-8f9b-4c9814ba9173.png)

**Comment by Aymkdn** [comment]
Thanks, it helps. I'll look at it on Monday.

**Comment by Saad1379** [comment]
@Aymkdn Thanks a lot.

**Comment by Aymkdn** [comment]
You put the `white-space:break-spaces` on the `H2` instead of the `P` tag…

It should be:
```html
<div>
  <h2 style="font-size:18px;font-weight:700;color:#0b6ab0;text-align:center;">Heading<h2>
  <p style="white-space:break-spaces">
  Lorem ipsum dolor, vero numquam
  molestiae quidem, dolorum.
  </p>
</div>
```

![image](https://user-images.githubusercontent.com/946315/173300359-47990b89-0b9f-4993-a196-350c87e2579b.png)

**Comment by Saad1379** [comment]
@Aymkdn You can see in your screenshot that there is an extra empty line between paragraph and heading in the pdf. This is the problem with `white-space:break-spaces;` property
![image](https://user-images.githubusercontent.com/16044705/173303579-b9a4b3f9-46cd-448f-bf62-7569be20964a.png)

**Comment by Aymkdn** [comment]
Your explanation let me thing your issue was with a text on several lines (see my example), but actually I think I now understand that your issue is with a text that is an unique sentence but displayed on several lines in the HTML code, and that should render on a single line in the PDF.

I just published the v2.4.0 that handles this case.

![image](https://user-images.githubusercontent.com/946315/173308799-4318beaf-4a03-4a22-9b7f-9fa6a37edfdb.png)
